ocaml-lsp-server/docs/ocamllsp/config.md

```diff
@@ -0,0 +1,21 @@
+# Configuration
+
+The ocamllsp support the folowing configuration. These configurations are sent through the [`didChangeConfiguration`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeConfiguration) notification.
+
+```ts
+interface config {
+  /**
+  * Enable Extended Hover
+  * @default false
+  * @since 1.16
+  */
+  extendedHover: { enable : boolean }
```

Comment:
would it be simpler to use boolean directly, instead of an object with `enable` key?
It would be similar to config that is used in VSCode:
```
"[ocaml]": {
  "editor.codeLens": false
}
```

Reply:
If in the future more options related to codelens were added then it would be inside the `codelens` object.

An example: adding codelens to show public values and types https://github.com/ocaml/ocaml-lsp/issues/1082

```json
{
  "codelens": {
    "enabled": true,
    "displayPublicValuesTypes": true
  }
}
```

This is just a pattern I've seen on some servers.